### DISCUSSION TITLE
How to enable hidpi scaling?

### DISCUSSION THREAD

**Discussion by Novined** [labwc/labwc#1157] [General]
Hello, can't find in documentation. On my 3k display elements are very small.

**Comment by stefonarch** [comment]
Setting it in "environment" file should work (I'm using lxqt-session settings for the same):

`export GDK_SCALE=1`

```
$ env|grep SCALE
GDK_SCALE=1
QT_SCALE_FACTOR=1
```

**Comment by Novined** [comment]
> Setting it in "environment" file should work (I'm using lxqt-session settings for the same):
> 
> `export GDK_SCALE=1`
> 
> ```
> $ env|grep SCALE
> GDK_SCALE=1
> QT_SCALE_FACTOR=1
> ```

Thank you, but it's not complete solution. How to do fractional scaling? And What to do with labwc cursor and elements?

**Reply by stefonarch** [reply]
Cursor size, title bar height and font size, menu sizes  and svg icons can be configured.
Fractional scaling works fine here with Qt.

**Reply by Consolatis** [reply]
Using the output scale should also handle cursor and menu sizes, same for the window title. Including when moving a window between different outputs that have a different scale. SVG icons and rounded corners will be upscaled as well but may end up a bit blurry, there is work ongoing to make them re-render for new scales to make them non blurry as well.

**Comment by Novined** [comment]
> Cursor size, title bar height and font size, menu sizes and svg icons can be configured. Fractional scaling works fine here with Qt.

But how to deal with GTK?

**Reply by stefonarch** [reply]
Server side decoration?

EDIT: I see,` GDK_SCALE=2 rednotebook` doesn't work
In firefox and thunderbird there are tweaks in `about:config`, for the rest I've no idea.

**Reply by Novined** [reply]
Found information about server side decoration only in development aspect. But KDE and gnome somehow deal with installed apps.

**Comment by Flrian** [comment]
I don't use a HiDPI display so I might be wrong, but shouldn't it be enough to set the scale for a monitor via for example `wlr-randr --output eDP-1 --scale 1.5` ?

**Comment by johanmalm** [comment]
We’ve got the beginnings of a PR for the website on this. Any takers for finishing it off?

https://github.com/labwc/labwc.github.io/pull/18

**Comment by Consolatis** [comment]
As written by @Flrian above, setting the output scale should be enough.
Although the results of this may not be great for external applications if it is a fraction like `1.5`, so if possible I would suggest using `2` or `3`. I assume that Qt and GTK toolkits pick that setting up and scale appropriately without requiring any additional environment variables.
The output scale can be set via tools like `wlr-randr` (CLI), `wdisplays` (GUI) or `kanshi` (daemon that sets it automatically and reacts to hotplug events).

Labwc currently doesn't support the fractional scaling protocol as we are using the last released wlroots version rather than its development branch.
